components/eam/src/physics/cam/hetfrz_classnuc_cam.F90

```diff
@@ -876,368 +894,386 @@ end subroutine hetfrz_classnuc_cam_save_cbaero
-      r_bc = 0.067e-6_r8 ! from emission size
-   end if
+   if (as_bc > 0._r8) then
+      total_cloudborne_aer_num(1) = as_bc/aermc_accum_total * aer_cb(num_accum) * num_m3_to_cm3 ! #/cm^3
+   endif
 
-   if (aer(ii,kk,dst_accum)*1.0e-3_r8 > 1.0e-30_r8 .and. dst1_num > 1.0e-3_r8) then
-      r_dust_a1 = ( 3._r8/(4*pi*specdens_dust)*aer(ii,kk,dst_accum)/(dst1_num*1.0e6_r8) )**(1._r8/3._r8)
-   else
-      r_dust_a1 = 0.258e-6_r8
-   end if
+   if (as_du > 0._r8) then
+      total_cloudborne_aer_num(2) = as_du/aermc_accum_total * aer_cb(num_accum) * num_m3_to_cm3 ! #/cm^3
+   endif
 
-   if (aer(ii,kk,dst_coarse)*1.0e-3_r8 > 1.0e-30_r8 .and. dst3_num > 1.0e-3_r8) then
-      r_dust_a3 = ( 3._r8/(4*pi*specdens_dust)*aer(ii,kk,dst_coarse)/(dst3_num*1.0e6_r8) )**(1._r8/3._r8)
-   else
-      r_dust_a3 = 1.576e-6_r8
-   end if
+
+   dmc_imm = aer_cb(dst_coarse)
+
+   aermc_coarse_total = aer_cb(ncl_coarse) + aer_cb(dst_coarse) + aer_cb(bc_coarse) + &
+                        aer_cb(pom_coarse) + aer_cb(soa_coarse) + aer_cb(mom_coarse)
+
+   if (dmc_imm > 0._r8) then
+      total_cloudborne_aer_num(3) = dmc_imm/aermc_coarse_total * aer_cb(num_coarse) * num_m3_to_cm3 ! #/cm^3
+   endif
+
+end subroutine calculate_cloudborne_aer_num                                                                                                                                    
+
+!====================================================================================================                                                                                                                                      
+subroutine calculate_mass_mean_radius(ncnst, aer, &
+                                      total_interstitial_aer_num, &
+                                      hetraer)
+
+   !***************************************************
+   ! calculate mass mean radius for BC and dust
+   !***************************************************
+
+   ! input
+   integer,  intent(in) :: ncnst
+   real(r8), intent(in) :: aer(ncnst) ! interstitial aerosol concentrations [kg/m^3]
+   real(r8), intent(in) :: total_interstitial_aer_num(hetfrz_aer_nspec) ! interstitial concentrations of BC and dust [#/cm^3]
+
+   ! output
+   real(r8), intent(out) :: hetraer(hetfrz_aer_nspec)                   ! BC and Dust mass mean radius [m]
+
+   ! local variables
+   real(r8) :: bc_num, dst1_num, dst3_num                ! number concentration [#/cm^3]
+   real(r8), parameter :: aermc_min_threshold  = 1.0e-30_r8
+   real(r8), parameter :: aernum_min_threshold = 1.0e-3_r8
+   real(r8), parameter :: r_bc_prescribed = 0.067e-6_r8
+   real(r8), parameter :: r_dust_a1_prescribed = 0.258e-6_r8
+   real(r8), parameter :: r_dust_a3_prescribed = 1.576e-6_r8
+
+   bc_num   = total_interstitial_aer_num(1)
+   dst1_num = total_interstitial_aer_num(2)
+   dst3_num = total_interstitial_aer_num(3)
+
+   hetraer(1) = r_bc_prescribed
+   hetraer(2) = r_dust_a1_prescribed
+   hetraer(3) = r_dust_a3_prescribed
```

Comment:
Would you please add a comment here that these variables are initialized using their prescribed values?

Reply:
comments added